packages/@react-aria/select/src/HiddenSelect.tsx

```diff
@@ -141,13 +141,19 @@ export function HiddenSelect<T>(props: HiddenSelectProps<T>): JSX.Element | null
       </div>
     );
   } else if (name) {
+    let data = selectData.get(state) || {};
+    // Use a hidden <input type="text"> rather than <input type="hidden">
+    // so that an empty value blocks HTML form submission when the field is required.
     return (
       <input
-        type="hidden"
+        type="text"
```

Comment:
This should only be the case if `validationBehavior="native"`. If `validationBehavior="aria"`, then form submission should not be blocked. We should continue to use `type="hidden"` in that case.